src/inventory/mod.rs

```diff
@@ -237,12 +375,33 @@ impl InventoryType {
             21 => InventoryType::Smoker,
             22 => InventoryType::CartographyTable,
             23 => InventoryType::Stonecutter,
-            _ => InventoryType::Chest(1),
-        }
+            _ => {
+                println!("Unhandled inventory type {id}");
+                return None;
+            }
+        })
     }
 
-    pub fn from_name(_name: String) -> Self {
-        InventoryType::Chest(1)
+    // Lookup a window type based on the inventory type strings used between
+    // 1.8 and 1.13.
+    pub fn from_name(name: &str, slot_count: u8) -> Option<Self> {
+        Some(match name {
+            "minecraft:anvil" => InventoryType::Anvil,
+            "minecraft:beacon" => InventoryType::Beacon,
+            "minecraft:brewing_stand" => InventoryType::BrewingStand,
+            "minecraft:chest" => InventoryType::Chest(slot_count / 9),
```

Comment:
We should probably check if the slot_count is divisible by 9 before assuming that it is and return None if it isn't divisible by 9 - or even panic if there is no way for the server to trick us into constructing an InventoryType. if there is a way for the server to make us construct an InventoryType, then we should probably just return None and log some error message into the console.